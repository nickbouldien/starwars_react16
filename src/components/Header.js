import React from 'react';
import { Link } from 'react-router-dom';
import { Menu, List, Button, Segment, Icon } from 'semantic-ui-react';

const Header = () => (
  <Menu inverted>
    <Menu.Item
      header
      name={'React 16 SW'}
      as={Link}
      to={"/"}
    />
    <Menu.Menu position={"right"}>
      <Menu.Item
        name={'about'}
        // active={activeItem === 'about'}
        as={Link}
        to={"/about"}
      />
      <Menu.Item
        name={'Github link'}
        as={'a'}
        href={"https://www.github.com/nickbouldien" } // TODO: make link to actual repo
        target='_blank'
        rel='noopener noreferrer'
      >
        <Icon name="github" size="large" />
      </Menu.Item>
    </Menu.Menu>
  </Menu>
);

export default Header;

