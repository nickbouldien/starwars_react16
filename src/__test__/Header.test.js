// Frag.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Link, MemoryRouter } from 'react-router-dom';
import Header from '../components/Header';
import setup from './setup';


describe('<Header /> component', () => {
  it('renders (shallow) the Header component', () => {
    const wrapper = shallow(<Header />);
    // console.log('wrapper', toJson(wrapper.children()));
    expect(wrapper.children().length).toBe(2);
    // expect(wrapper.find('a').length).toBe(3);
  });

  test('Header matches snapshot', () => {
    const component = mount(<MemoryRouter><Header /></MemoryRouter> );
    console.log('link: ', toJson(component.find('a')));
    expect(component.find('a').first().text()).toBe('React 16 SW');

    // can't do the snapshot since they contain keys which change each time
    // expect(toJson(component)).toMatchSnapshot();
  });
});