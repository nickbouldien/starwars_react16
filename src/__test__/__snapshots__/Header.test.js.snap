// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component Header matches snapshot 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "dfg9nt",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "dfg9nt",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Header>
      <Menu
        inverted={true}
      >
        <div
          className="ui inverted menu"
        >
          <MenuItem
            as={[Function]}
            header={true}
            name="React 16 SW"
            to="/"
          >
            <Link
              className="header item"
              onClick={[Function]}
              replace={false}
              to="/"
            >
              <a
                className="header item"
                href="/"
                onClick={[Function]}
              >
                React 16 SW
              </a>
            </Link>
          </MenuItem>
          <MenuMenu
            position="right"
          >
            <div
              className="right menu"
            >
              <MenuItem
                as={[Function]}
                name="about"
                to="/about"
              >
                <Link
                  className="item"
                  onClick={[Function]}
                  replace={false}
                  to="/about"
                >
                  <a
                    className="item"
                    href="/about"
                    onClick={[Function]}
                  >
                    About
                  </a>
                </Link>
              </MenuItem>
              <MenuItem
                as="a"
                href="https://www.github.com/nickbouldien"
                name="Github link"
                rel="noopener noreferrer"
                target="_blank"
              >
                <a
                  className="item"
                  href="https://www.github.com/nickbouldien"
                  onClick={[Function]}
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <Icon
                    as="i"
                    name="github"
                    size="large"
                  >
                    <i
                      aria-hidden="true"
                      className="github large icon"
                    />
                  </Icon>
                </a>
              </MenuItem>
            </div>
          </MenuMenu>
        </div>
      </Menu>
    </Header>
  </Router>
</MemoryRouter>
`;
